packages/azure-functions/ProcessTrustedFile/helpers/process-geospatial-land-boundary.js

```diff
@@ -1,24 +1,47 @@
 import buildSignalRMessage from '../../Shared/build-signalr-message.js'
 import { CoordinateSystemValidationError, ValidationError } from '@defra/bng-errors-lib'
+import { blobStorageConnector } from '@defra/bng-connectors-lib'
 import { processLandBoundary } from '@defra/bng-geoprocessing-service'
 
+const GEOJSON_FILE_EXTENSION = '.geojson'
+
 export default async function (context, config) {
-  const processedFileLocation =
-   `${config.fileConfig.fileDirectory}/${config.fileConfig.filename}.geojson`
-  // Use the GDAL virtual file system to convert a geospatial land boundary uploaded
-  // by a user to GeoJSON.
+  // Defect - BNGP-1711.
+  // Add a timestamp to GeoJSON files created by GDAL so that the filename is unique.
+  // If an attempt is made to open a file with the same name as an existing file generated by GDAL,
+  // an error occurs until the function app is restarted.
+  const isGeoJsonFile = config.fileConfig.fileExtension === GEOJSON_FILE_EXTENSION
+  const tmpGeoJsonBlobName =
+   `${config.fileConfig.fileDirectory}/${config.fileConfig.filename}${new Date().toISOString()}}${GEOJSON_FILE_EXTENSION}`
```

Comment:
Doesn't break the logic but there is an extra } in the tmpGeoJsonBlobName string eg `geopackage-land-boundary-43262022-12-05T14:23:36.858Z}.geojson`

Reply:
Refactored